core/src/main/java/org/mule/runtime/core/api/rx/Exceptions.java

```diff
@@ -153,4 +155,29 @@ public static MuleException rxExceptionToMuleException(Throwable throwable)
     }
   }
 
+  /**
+   * Propagate an exception through streams even if it's considered fatal by Reactor.
+   *
+   * @param t throwable.
+   * @return exception wrapped by Reactor, after possibly wrapping with {@link MuleFatalException}.
+   */
+  public static RuntimeException propagateFatal(Throwable t) {
+    return propagate(wrapFatal(t));
+  }
+
+  /**
+   * Wrap an exception with {@link MuleFatalException} so that Reactor can handle it safely instead of always bubbling it up.
+   *
+   * @param t throwable.
+   * @return possibly wrapped exception.
+   */
+  public static Throwable wrapFatal(Throwable t) {
+    if (t instanceof LinkageError) {
+      return new MuleFatalException(t);
+    } else if (t instanceof VirtualMachineError) {
+      return new MuleFatalException(t);
+    } else {
```

Comment:
They are private so we can't catch them. We could always try to unwrap them, though. But if it was a ReactiveException we wouldn't be wrapping it again in wrapFatal, only in propagateWrappingFatal.

Would it be OK to just use propagateWrappingFatal instead of wrapFatal?